### PR TITLE
Rename GDNative to GDExtension

### PR DESCRIPTION
**Pull request by groud**
godot-cpp equivalent of https://github.com/godotengine/godot/pull/69718

List of renames (probably missing a few of them but that should be sufficient for covering 99% of PR rebases):
```
GDExtension -> GDNative
GDNATIVE -> GDEXTENSION
gdextension -> gdnative
ExtensionExtension ->Extension (for where there was GDNativeExtension)
EXTENSION_EXTENSION ->EXTENSION (for where there was GDNATIVE_EXTENSION)
gdnlib -> gdextension
gdn_interface -> gde_interface
gdni -> gde_interface
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # CMAKE_BUILD_TYPE:			Compilation target (Debug or Release defaults to Debug)
 #
 # godot-cpp cmake arguments
-# GODOT_HEADERS_DIR:		This is where the gdnative include folder is (godot_source/modules/gdnative/include)
+# GODOT_HEADERS_DIR:		This is where the gdextension include folder is (godot_source/modules/gdextension/include)
```
Comment (akien-mga) [inline CMakeLists.txt:5]: That's obsolete.

**Review by groud — COMMENTED** [review]
Approved in PR meeting. Some modified names will need to be reviewed before 4.0, to check, at least in gdextension_interface.h if everything makes sense.

Note to myself: list the sed commands to execute for everybody to update their code.

**Comment by akien-mga** [comment]
> Note to myself: list the sed commands to execute for everybody to update their code.

Doesn't need to be the exhaustive list of sed commands that would work out of the box, just a list of patterns that have been renamed:
- GDNativeExtension -> GDExtension
- GDNative -> GDExtension
- NativeExtension -> GDExtension
- etc.

**Comment by groud** [comment]
Yeah, I did something like:
- GDExtension -> GDNative
- GDNATIVE -> GDEXTENSION
- gdextension -> gdnative
- ExtensionExtension ->Extension (for where there was GDNativeExtension)
- gdnlib -> gdextension
- gdn_interface -> gde_interface,
I might miss some of them but that's pretty much it.

**Comment by BastiaanOlij** [comment]
I'm for merging this, looks good @groud

**Comment by BastiaanOlij** [comment]
> Yeah, I did something like:
> 
>     * GDExtension -> GDNative
> 
>     * GDNATIVE -> GDEXTENSION
> 
>     * gdextension -> gdnative
> 
>     * ExtensionExtension ->Extension (for where there was GDNativeExtension)
> 
>     * EXTENSION_EXTENSION ->EXTENSION (for where there was GDNATIVE_EXTENSION)
> 
>     * gdnlib -> gdextension
> 
>     * gdn_interface -> gde_interface,
>       I might miss some of them but that's pretty much it.

Was the idea to have this information in the commit message? I can't remember exactly what the discussion was during the GDExtension meeting

**Comment by groud** [comment]
> Was the idea to have this information in the commit message? I can't remember exactly what the discussion was during the GDExtension meeting

That's not what I understood. But it seems like a good idea.

**Review by akien-mga — APPROVED** [review]
Seems good to me.

**Comment by akien-mga** [comment]
> > Was the idea to have this information in the commit message? I can't remember exactly what the discussion was during the GDExtension meeting
> 
> That's not what I understood. But it seems like a good idea.

I meant having a list in the PR's OP, but having it in the commit message itself wouldn't hurt.

**Comment by groud** [comment]
> I meant having a list in the PR's OP, but having it in the commit message itself wouldn't hurt.

Ok, I'll add it.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**